src/tests/system/tests/test_files.py

```diff
@@ -0,0 +1,120 @@
+"""
+Files test provider cases
+
+:requirement: IDM-SSSD-REQ :: SSSD is default for local resolution
+"""
+
+from __future__ import annotations
+
+import time
+
+import pytest
+from sssd_test_framework.roles.client import Client
+from sssd_test_framework.topology import KnownTopology
+
+
+@pytest.mark.builtwith("files-provider")
+@pytest.mark.topology(KnownTopology.Client)
+def test_files__getent_does_not_handle_root(client: Client):
+    """
+    :title: Getent call doesnt work on root, when service specified as "sss"
+    :setup:
+        1. Enable files domain
+        2. Start SSSD
+    :steps:
+        1. getent passwd -s sss root
+    :expectedresults:
+        1. Call failed
+    :customerscenario: False
+    """
+    client.sssd.sssd["enable_files_domain"] = "true"
+    client.sssd.start()
+
+    result = client.tools.getent.passwd("root", service="sss")
+    assert result is None, "Getent call was successful, which is not expected"
+
+
+@pytest.mark.builtwith("files-provider")
+@pytest.mark.topology(KnownTopology.Client)
+def test_files__simple_getent(client: Client):
+    """
+    :title: Simple getent call
+    :setup:
+        1. Add local user "user1"
+        2. Enable files domain
+        3. Start SSSD
+    :steps:
+        1. getent passwd -s sss user1
+        2. Check uid of result
+    :expectedresults:
+        1. Call was successful
+        2. Uid is correct
+    :customerscenario: False
+    """
+    client.local.user("user1").add(uid=10001)
+    client.sssd.sssd["enable_files_domain"] = "true"
+    client.sssd.start()
+
+    result = client.tools.getent.passwd("user1", service="sss")
+    assert result is not None, "Getent failed"
+    assert result.uid == 10001, "Uid is not correct"
+
+
+@pytest.mark.builtwith("files-provider")
+@pytest.mark.topology(KnownTopology.Client)
+def test_files__enumeration(client: Client):
+    """
+    :title: Files provider should not enumerate
+    :setup:
+        1. Enable files domain
+        2. Start SSSD
+    :steps:
+        1. getent passwd -s sss without specified user
+    :expectedresults:
+        1. Output is empty
+    :customerscenario: False
+    """
+    client.sssd.sssd["enable_files_domain"] = "true"
+    client.sssd.start()
+
+    result = client.host.ssh.run("getent passwd -s sss")
+    assert not result.stdout
```

Comment:
Hi,

is there a reason why you do not use `assert result is None` similar to the other converted tests? I see that the original test is explicitly checking stdout_text but, if possible, just checking `result` for `None` would be more consistent here.

bye,
Sumit

Reply:
Hi,
the reason is that here I call `getent` command directly through `ssh.run()`. That means that result is SSHProccesResult and not PasswdEntry. And the reason why it is called directly is that our getentUtils utility does not support calls without username, which is what I am doing here.

Reply:
Hi,

thanks for the explanation.

bye,
Sumit